src/uct/ib/base/ib_md.c

```diff
@@ -581,7 +581,7 @@ ucs_status_t uct_ib_memh_alloc(uct_ib_md_t *md, size_t length,
     memh->rkey = UCT_IB_INVALID_MKEY;
 
     if ((mem_flags & UCT_MD_MEM_FLAG_NONBLOCK) && (length > 0) &&
-        (md->reg_nonblock_mem_types & UCS_BIT(UCS_MEMORY_TYPE_HOST))) {
+        (md->reg_nonblock_mem_types != 0)) {
```

Comment:
can we do this according to the mem type of the specific buffer? otherwise no point to check reg_nonblock_mem_types at all..

Reply:
agree (this is existing issue between host and cuda mem_type). I was assuming:
- updated check remains a first level of sanity check
- mem_type lookup in IB could be costly and not already done at all
- caller (aka ucp) is already taking care in practice to make sane calls

There is the related #10079. To address your comment, do you see a way to check mem type in IB?

Reply:
removed the check as discussed